intersection_transit_maneuvering/src/intersection_transit_maneuvering.cpp

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (C) 2021 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include <ros/ros.h>
+#include <string>
+#include <algorithm>
+#include <memory>
+#include <boost/uuid/uuid_generators.hpp>
+#include <boost/uuid/uuid_io.hpp>
+#include <lanelet2_core/geometry/Point.h>
+#include <trajectory_utils/trajectory_utils.h>
+#include <trajectory_utils/conversions/conversions.h>
+#include <sstream>
+#include <carma_utils/containers/containers.h>
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+#include <Eigen/LU>
+#include <Eigen/SVD>
+#include <intersection_transit_maneuvering.h>
+
+
+using oss = std::ostringstream;
+
+namespace intersection_transit_maneuvering
+{
+    IntersectionTransitManeuvering::IntersectionTransitManeuvering(carma_wm::WorldModelConstPtr& wm, PublishPluginDiscoveryCB plugin_discovery_publisher,
+                                                                     std::shared_ptr<Interface> obj)
+    {        
+        plugin_discovery_msg_.name = "IntersectionTransitManeuvering";
+        plugin_discovery_msg_.versionId = "v1.0";
+        plugin_discovery_msg_.available = true;
+        plugin_discovery_msg_.activated = false;
+        plugin_discovery_msg_.type = cav_msgs::Plugin::TACTICAL;
+        plugin_discovery_msg_.capability = "tactical_plan/plan_trajectory";
+
+        object = obj;
+    }
+
+    bool IntersectionTransitManeuvering::plan_trajectory_cb(cav_srvs::PlanTrajectoryRequest& req, cav_srvs::PlanTrajectoryResponse& resp)
+    {
+        ros::WallTime start_time = ros::WallTime::now();  // Start timeing the execution time for planning so it can be logged
+
+        std::vector<cav_msgs::Maneuver> maneuver_plan;
+        for(size_t i = req.maneuver_index_to_plan; i < req.maneuver_plan.maneuvers.size(); i++)
+        {
+            if(req.maneuver_plan.maneuvers[i].type == cav_msgs::Maneuver::INTERSECTION_TRANSIT_STRAIGHT ||
+            req.maneuver_plan.maneuvers[i].type == cav_msgs::Maneuver::INTERSECTION_TRANSIT_LEFT_TURN || 
+            req.maneuver_plan.maneuvers[i].type == cav_msgs::Maneuver::INTERSECTION_TRANSIT_RIGHT_TURN )
+            {
+                maneuver_plan.push_back(req.maneuver_plan.maneuvers[i]);
+                resp.related_maneuvers.push_back(i);
+            }
+            else
+                {
+                    throw std::invalid_argument("Invalid Maneuver Type");
```

Comment:
You should break here instead of throw an exception, because there may be planned maneuvers the intersection which will be passed to different plugins after your plugin completes. So you want to break out of this loop once you have accounted for your related maneuvers. Throwing an exception by contrast would shutdown the whole system. 

It would be appropriate to throw an exception only if req.maneuver_plan.maneuvers[req.maneuver_index_to_plan] was not an intersection maneuver.